app/src/userpages/modules/userPageStreams/actions.js

```diff
@@ -320,9 +321,9 @@ export const createStream = (options: { name: string, description: ?string }) =>
 export const updateStream = (stream: Stream) => (dispatch: Function) => {
     dispatch(updateStreamRequest())
     return services.putStream(stream.id, stream)
-        .then(handleEntities(streamSchema, dispatch))
-        .then((id) => {
-            dispatch(updateStreamSuccess(id))
+        .then(() => handleEntities(streamSchema, dispatch)(stream))
+        .then(() => {
+            dispatch(updateStreamSuccess(stream))
```

Comment:
You could have also changed the parameter for `updateStreamSuccess` to an id, it's not really used for anything.